fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/PlayerBlockBreakEvents.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.player;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class PlayerBlockBreakEvents {
+	private PlayerBlockBreakEvents() { }
+
+	/**
+	 * Callback before a block is broken.
+	 * Only called on the server, however updates are synced with the client.
+	 *
+	 * <p>If any listener cancels a block breaking action, that block breaking
+	 * action is cancelled and {@link CANCELED} event is fired. Otherwise, the
+	 * {@link AFTER} event is fired.</p>
+	 */
+	public static final Event<Before> BEFORE = EventFactory.createArrayBacked(Before.class,
+			(listeners) -> (world, player, pos, state, entity) -> {
+				for (Before event : listeners) {
+					boolean result = event.beforeBlockBreak(world, player, pos, state, entity);
+
+					if (!result) {
+						return false;
+					}
+				}
+
+				return true;
+			}
+	);
+
+	/**
+	 * Callback after a block is broken.
+	 *
+	 * <p>Called on the Server only.</p>
```

Comment:
```suggestion
	 * <p>Only called on a logical server.
```

You can omit the closing `</p>` in the style but it isn't required of course